core/templates/dev/head/pages/topic_editor/TopicEditorStateService.js

```diff
@@ -234,21 +250,26 @@ oppia.factory('TopicEditorStateService', [
           if (newIndex !== null) {
             _newSubtopicPages.splice(newIndex, 1);
             for (var i = 0; i < _subtopicPages.length; i++) {
-              var newSubtopicId = parseInt(_subtopicPages[i].getId().slice(13));
+              var newSubtopicId = _getSubtopicIdFromSubtopicPageId(
+                _subtopicPages[i].getId());
               if (newSubtopicId > subtopicId) {
                 newSubtopicId--;
-                _subtopicPages[i].setId(topicId + '-' + newSubtopicId);
+                _subtopicPages[i].setId(
+                  _getSubtopicPageId(topicId, newSubtopicId));
               }
             }
             for (var i = 0; i < _newSubtopicPages.length; i++) {
-              var newSubtopicId = parseInt(
-                _newSubtopicPages[i].getId().slice(13));
+              var newSubtopicId = _getSubtopicIdFromSubtopicPageId(
+                _newSubtopicPages[i].getId());
               if (newSubtopicId > subtopicId) {
                 newSubtopicId--;
-                _newSubtopicPages[i].setId(topicId + '-' + newSubtopicId);
+                _newSubtopicPages[i].setId(
+                  _getSubtopicPageId(topicId, newSubtopicId));
               }
             }
           }
+        } else {
+          throw Error('The given subtopic page doesn\'t exist');
```

Comment:
Note: better to do:

```
if (index === null)
  throw Error

// rest of your logic here
```

That way you remove an extra level of indentation from the "main" part of the code.

Reply:
Done.